docs/examples/rag_weaviate.ipynb

```diff
@@ -0,0 +1,1092 @@
+{
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Ag9kcX2B_atc"
+      },
+      "source": [
+        "# Performing RAG over PDFs with Weaviate and Docling\n",
+        "## A recipe 🧑‍🍳 🐥 💚\n",
+        "[![Open in Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/your-username/your-repo/blob/main/notebook.ipynb)\n",
+        "\n",
+        "By Mary Newhauser, MLE @ Weaviate\n",
+        "\n",
+        "This is a code recipe that uses [Weaviate](https://weaviate.io/) to perform RAG over PDF documents parsed by [Docling](https://ds4sd.github.io/docling/).\n",
+        "\n",
+        "In this notebook, we accomplish the following:\n",
+        "* Parse the top machine learning papers on [arXiv](https://arxiv.org/) using Docling\n",
+        "* Perform hierarchical chunking of the documents using Docling\n",
+        "* Generate text embeddings with OpenAI\n",
+        "* Perform RAG using [Weaviate](https://weaviate.io/developers/weaviate/search/generative)\n",
+        "\n",
+        "To run this notebook, you'll need:\n",
+        "* An [OpenAI API key](https://platform.openai.com/docs/quickstart)\n",
+        "* Access to GPU/s\n",
+        "\n",
+        "Note: For best results, please use **GPU acceleration** to run this notebook. Here are two options for running this notebook:\n",
+        "1. **Locally on a MacBook with an Apple Silicon chip.** Converting all documents in the notebook takes ~2 minutes on a MacBook M2 due to Docling's usage of MPS accelerators.\n",
+        "2. **Run this notebook on Google Colab.** Converting all documents in the notebook takes ~8 mintutes on a Google Colab T4 GPU."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "4YgT7tpXCUl0"
+      },
+      "source": [
+        "### Install Docling and Weaviate client\n",
+        "\n",
+        "Note: If Colab prompts you to restart the session after running the cell below, click \"restart\" and proceed with running the rest of the notebook."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 1,
+      "metadata": {
+        "collapsed": true,
+        "id": "u076oUSF_YUG"
+      },
+      "outputs": [],
+      "source": [
+        "%%capture\n",
+        "%pip install docling==\"2.7.0\"\n",
+        "%pip install -U weaviate-client==\"4.9.4\"\n",
```

Comment:
I would rather not pin the versions so strictly, e.g. you can instead do sth like `docling~=2.7.0` and `weaviate-client~=4.9.4` for capturing [compatible releases](https://packaging.python.org/en/latest/specifications/version-specifiers/#compatible-release).